check/src/kindcheck.rs

```diff
@@ -143,7 +120,7 @@ impl<'a> KindCheck<'a> {
     fn builtin_kind(&self, typ: BuiltinType) -> RcKind {
         match typ {
             BuiltinType::String | BuiltinType::Byte | BuiltinType::Char | BuiltinType::Int |
-            BuiltinType::Float | BuiltinType::Unit => self.type_kind(),
+            BuiltinType::Float | BuiltinType::Unit | BuiltinType::Type => self.type_kind(),
             BuiltinType::Array => self.function1_kind(),
             BuiltinType::Function => self.function2_kind(),
```

Comment:
If we did have universes, I'm not sure how we would handle them here and in other places. We'd need to figure out structurally where the largest universe is within a type... I think?